libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -625,6 +633,28 @@ class MapboxNavigationTest {
         mapboxNavigation.onDestroy()
     }
 
+    @Test
+    fun `verify tile config host`() {
+        ThreadController.cancelAllUICoroutines()
+        val slot = slot<TilesConfig>()
+        every {
+            NavigationComponentProvider.createNativeNavigator(any(), any(), capture(slot))
+        } returns navigator
+        val options = navigationOptions.toBuilder()
+            .onboardRouterOptions(
+                OnboardRouterOptions.Builder()
+                    .tilesUri(URI.create("https://api.mapbox.com/route-tiles/v2/dataset/profile"))
```

Comment:
This is an incorrect test. `tilesUri` should not accept `"https://api.mapbox.com/route-tiles/v2/dataset/profile`, only `https://api.mapbox.com`. The dataset and profile have separate builder methods in the `OnboardRouterOptions.Builder`.

Reply:
Maybe we should change the name of the variable from `tilesUri` to `tilesBaseUrl`?

Reply:
@LukasPaczos according to [this](https://github.com/mapbox/1tap-android/issues/705#issuecomment-732455965) we need update `routing_tiles_endpoint` . 
and it will have a tale, for example
`https://api.mapbox.com/route-tiles/v2/mapbox-single-guidance-here/driving-traffic`
in 1Tap we do not check it and pass to NavSDK, so we pass a full URI and it's invalid from NN point of view.
We have 2 options:
- handle it on 1Tap side (no guarantee that another customer will not send a full URI)
- handle on SDK side

Reply:
I think we should handle it on the 1TAP side and split up the URL into the correct pieces and pass them to the builder. This is a Nav SDK v2 breaking change and it should be handled by the consumer and noted in the migration guide.

We might want to add additional logs when parsing the options if we detect that the schema doesn't fit.